joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForGenDeclarationCreator.scala

```diff
@@ -12,129 +12,115 @@ import scala.util.Try
 
 trait AstForGenDeclarationCreator(implicit withSchemaValidation: ValidationMode) { this: AstCreator =>
   def astForGenDecl(genDecl: ParserNodeInfo): Seq[Ast] = {
-    genDecl.json(ParserKeys.Tok).str match {
-      case "import" => astForImport(genDecl)
-      case "const"  => List[Ast]()
-      case "type"   => astForTypeSpec(genDecl)
-      case "var"    => astForValueSpec(genDecl)
-
-    }
-  }
-
-  def astForTypeSpec(genDecl: ParserNodeInfo): Seq[Ast] = {
     genDecl
       .json(ParserKeys.Specs)
       .arr
       .map(createParserNodeInfo)
-      .flatMap { typeSpec =>
-        typeSpec.node match {
-          // TODO: Add support for member variables and methods
-          case TypeSpec =>
-            val nameNode          = typeSpec.json(ParserKeys.Name)
-            val typeNode          = typeSpec.json(ParserKeys.Type)
-            val astParentType     = methodAstParentStack.head.label
-            val astParentFullName = methodAstParentStack.head.properties("FULL_NAME").toString
-            val typeDeclNode_ =
-              typeDeclNode(
-                typeSpec,
-                nameNode(ParserKeys.Name).str,
-                x2cpg.Defines.DynamicCallUnknownFullName, // TODO: Fill in fullName
-                parserResult.filename,
-                typeSpec.code,
-                astParentType,
-                astParentFullName
-              )
-
-            addModifier(typeDeclNode_, nameNode(ParserKeys.Name).str)
-
-            // TODO: Fill in typeFullName
-            val typeRefNode_ =
-              typeRefNode(typeSpec, s"type ${nameNode(ParserKeys.Name).str} struct", Defines.anyTypeName)
-
-            diffGraph.addEdge(methodAstParentStack.head, typeDeclNode_, EdgeTypes.AST)
-            Seq(Ast(typeRefNode_))
-          case _ =>
-            Seq(Ast())
-        }
+      .flatMap { genDeclNode =>
+        genDecl.json(ParserKeys.Tok).str match
+          case "import" => astForImport(genDeclNode)
+          case "const"  => List[Ast]()
+          case "type"   => astForTypeSpec(genDeclNode)
+          case "var"    => astForValueSpec(genDeclNode)
+          case _        => List[Ast]()
       }
+      .toSeq
+  }
+
+  def astForTypeSpec(typeSpec: ParserNodeInfo): Seq[Ast] = {
+    typeSpec.node match {
+      // TODO: Add support for member variables and methods
```

Comment:
We can move this Switch case statement as well to the caller function and remove the instead of having the switch case on `genDecl.json(ParserKeys.Tok)`